Option to select the scale of the Y axis

jasoncalabrese opened the pull request:
<img width="288" alt="screen shot 2015-10-19 at 7 23 27 pm" src="https://cloud.githubusercontent.com/assets/751143/10596379/5cd29932-7698-11e5-8fb1-d9a2ed86a820.png">

<img width="1271" alt="screen shot 2015-10-19 at 7 23 44 pm" src="https://cloud.githubusercontent.com/assets/751143/10596380/5ea5138e-7698-11e5-88f5-1782db67a296.png">

jasoncalabrese commented:
@kenstack  @channemann @bewest

bewest commented:
Nice.

jasoncalabrese commented:
@Fokko think you had a version of this before also

MilosKozak commented:
Describe both options in readme

MilosKozak commented:
Ticks should be different for both types. You may take it from reports

jasoncalabrese commented:
good idea for the readme, suggestions for the text?   @oamyoamy?

the ticks aren't customizable yet, so setting good default for the linear scale make sense

MilosKozak commented:
https://github.com/nightscout/cgm-remote-monitor/blob/master/lib/report_plugins/daytoday.js#L93

jasoncalabrese commented:
:+1:

jasoncalabrese commented:
@MilosKozak I refactored the ticks some so they are shared between the main chart and day-to-day.  Would be nice to mix in the targets with the linear ticks.

MilosKozak commented:
:+1:

MilosKozak commented:
add  custom targets to linear too. I think sorting is not necessary  so you can add to the end.
you will have to move custom targets out of ticks.js or pass them as a params. in reports are customisable ....

oamyoamy commented:
Suggested readme info:   `SCALE_Y` (`log`) - The type of scaling used for the Y axis of the charts system wide. Set to 'linear' for chart scaling with equidistant tick marks. (Linear or log scaling can be altered locally via settings panel in the browser.)

channemann commented:
Good stuff, thanks for the mention

jasoncalabrese commented:
@MilosKozak I added support for the option used in day-to-day with the new ticks module, the 40 / 2 tick is getting cropped by the open top line with linear scaling.  I tried adjusting it, but wasn't working maybe you have an idea.  If we can't find a quick fix maybe if can be fixed later.

@oamyoamy I updated the readme for the Log/Linear options based on your suggestion and added a little more.  You could probably come up with something better.

jasoncalabrese commented on lib/client/ticks.js:
```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+function prepare (client, opts) {
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1150973166/issues/source?bid=2452379&fileBranchId=2909529#l3)

jasoncalabrese replied:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1156649228/issues/source?bid=2452379&fileBranchId=2909529#l3)

jasoncalabrese commented on lib/client/ticks.js:
```diff
@@ -0,0 +1,81 @@
+  }
+}
+
+function prepareLinear (client, opts) {
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['opts' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1159517207/issues/source?bid=2452379&fileBranchId=2909529#l50)

jasoncalabrese replied:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['opts' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1159572931/issues/source?bid=2452379&fileBranchId=2909529#l50)

jasoncalabrese commented:
A new option for dynamic ranges with the log scale
<img width="1269" alt="screen shot 2015-10-21 at 8 15 35 pm" src="https://cloud.githubusercontent.com/assets/751143/10656129/892ce14c-7830-11e5-99ae-3dbffc32a4e8.png">

MilosKozak commented:
MAYBE with dynamic scale would be nice to adjust basal scale too. to not overlap but use 100% of space
https://github.com/nightscout/cgm-remote-monitor/blob/dev/lib/client/chart.js#L209
https://github.com/nightscout/cgm-remote-monitor/pull/1222/files#diff-6cf944f042d6214695a022aad3d04f39R46
